rskj-core/src/main/java/org/ethereum/core/BlockHeader.java

```diff
@@ -496,6 +514,20 @@ public String getShortHashForMergedMining() {
         return hashForMergedMining;
     }
 
+    private byte[] getHashRootForMergedMining(byte[] leftHash) {
+        if ((ummRoot.length != UMM_LEAVES_LENGTH) && (ummRoot.length != 0)){
```

Comment:
This method should not be called when ummRoot.length ==0. It's better to throw an exception in that case.